ccp/lib/processors/pipeline_information/builds.py

```diff
@@ -67,6 +68,56 @@ def __init__(
                 token_from_mount=token_from_mount
             )
 
+    def get_unlisted_info_from_core_jenkins(
+            self, ordered_job_list, build_number, keys
+    ):
+        """
+        Gets unlisted information from core jenkins
+        :param ordered_job_list: The ordered list of jobs, with parents,
+        followed by children
+        :type ordered_job_list list
+        :param build_number: The number of the build whose information you want
+        :type build_number str
+        :param keys: The list containing jenkins keys whose information you want
+        :type keys dict
+        :return:
+        """
+        data = {}
+        try:
+            response = self.jenkins_core_client.get_build_info(
+                job_ordered_list=ordered_job_list,
+                build_number=build_number
+            )
+            # build result/status possible values = SUCCESS/FAILURE
+            data["RESULT"] = response.get("result")
+
+            # now parsing other details
+            actions = response.get("actions")
+            # actions contains a list of dictionaries
+
+            for each in actions:
+                # here each might be empty dict too
+                if not each.get("_class"):
+                    continue
+
+                if each.get("_class") == "hudson.model.ParametersAction":
+                    # parameters is a list of dict
+                    for param in each.get("parameters"):
+                        data[param["name"]] = param["value"]
+        except KeyError as e:
+            print ("Error parsing build details.")
+            raise e
+
+        # check if required_fields are in build_info
+        if not set(keys).issubset(data.keys()):
+            raise InformationNotInJenkinsError(
+                "Could not retrieve required field(s) from build details."
+                "Missing field(s) {}".format(
+                    list(set(keys).difference(data.keys()))
+                )
+            )
+        return data
```

Comment:
I think you should only return the keys as requested.

here, you are checking if `data` is super set of `keys`, if yes, returning the data, which might have additional info.

which is fine as it won't break, but it would be unnecessary back and forth of additional data.

say, I just want to grab a single parameter, while, you give me 10.

I would say, you match the requested keys from the data and return only that.